### PR TITLE
Setup Gatsby and add coming soon page for the 2019 site

### PR DESCRIPTION
**Pull request by RobertWSaunders**
## Proposed Change

Fixes: #30

Prepare the repository for the 2019 QHacks website. You can view a list of detailed changes but the main change here is that this year we will be utilizing [Gatsby](https://www.gatsbyjs.org/) to create our website. Utilizing Gatsby will give us performance benefits, use new technologies, continue to use React like our dashboard, utilize the same tools as the dashboard (Jest, Coveralls, Prettier, ESLint, etc), and more easily fetch data from our API's for dynamic content on our static site!

In addition to using Gatsby, this change adds a simple "Coming Soon" page to have up on the website while we build the new website. Also, not directly tied to this change is our new hosting provider Netlify, so anytime we make a PR to master it will automatically update the website. The new "Coming Soon" page will act as the change to test that deployment when we make a new release following this PR merging. 

I realize this change is a massive one and normally we would not allow this big of a change, that said with one of the main focuses for the tech team this year being documentation, general clean up, consistency, and specifically improving our codebase by having projects with similar tech stacks this change takes a large step in the right direction.

### How did you do this?

See the list of changes below!

### Why are you choosing this approach?

See the reasons above, namely consistency and improve our tech stack!

### Any open questions you want to ask code reviewers?

I left some comments in the diff, otherwise nope! 😃 

### List of changes:

  - Sets the repo up to be a Gatsby project!
  - Adds better documentation to the README.
  - Copies over all tools used in the dashboard that apply, namely:
    - Jest
    - Coveralls
    - ESLint
    - Prettier
  - Updates the GitHub templates to be consistent with the dashboard.
  - Brings over things like the License and Code of Conduct from the dashboard.
  - Creates a new "Coming Soon" page to be released until we have our new site built.
  - Copies over all the CI/CD configuration from the dashboard.
  - Adds code owners to the repo.
  - Adds a couple of tests to satisfy the Jest no test rule!

### Picture of Change:

screen shot 2018-09-10 at 6 56 42 pm

## Pull Request Checklist:

  - [x] My submission passes all tests.
  - [x] I have lint my code locally prior to submission.
  - [x] I have written new tests for my core changes (where applicable).
  - [x] I have referenced all useful information (issues, tasks, etc) in the pull request.

### CODE REVIEW

**Comment by RobertWSaunders** [comment]
Deploy preview for *qhacks-website* ready!

Built with commit f843c563aacbff668bad7cae3dfb6a1b12156218

https://deploy-preview-88--qhacks-website.netlify.com

File: src/__tests__/pages/Index/index.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+import IndexPage from "../../../pages/index";
+import { shallow } from "enzyme";
```
Comment (joeytepp) [inline src/__tests__/pages/Index/index.test.js]: Nice choice, I like the consistency with the dashboard 👍

File: .editorconfig [hunk 2]
```diff
@@ -2,7 +2,7 @@ root = true
 
 [*]
 charset = utf-8
-indent_style = tab
+indent_style = space
```
Comment (joeytepp) [inline .editorconfig]: 👌👌👌👌👌👌👌👌👌👌👌👌

File: src/__tests__/pages/Index/index.test.js (hunk 1 above)
Comment (RobertWSaunders) [reply src/__tests__/pages/Index/index.test.js]: Yeee!

File: .editorconfig (hunk 2 above)
Comment (RobertWSaunders) [reply .editorconfig]: 🔥

**Pull request merged**